src/editor/Neovim/BufferEvents.re

```diff
@@ -0,0 +1,140 @@
+open Oni_Core;
+open Types;
+
```

Comment:
I split out the functionality relating to building up the buffer list as it seemed to me to very quickly fill up `Oni2` and the functionality wasn't really coupled. I wasn't really sure  what to name it, I went with BufferEvents because I imagine there will be similar functionality required for dealing with buffers being closed or renamed etc. so this could basically be a module to deal with data transformation associated with those events.

Reply:
Update renamed this to `NeovimBuffer` as it is essentially a buffer to handle interactions with `neovim` buffers